contracts/StablePriceOracle.sol

```diff
@@ -9,64 +9,75 @@ interface DSValue {
-     * @param _usdOracle The address of the price oracle to use.
-     */
-    function setOracle(DSValue _usdOracle) public onlyOwner {
-        usdOracle = _usdOracle;
-        emit OracleChanged(address(_usdOracle));
+    function price(string calldata name, uint expires, uint duration) external view returns(uint) {
+        uint len = name.strlen();
+        if(len > rentPrices.length) {
+            len = rentPrices.length;
+        }
+        require(len > 0);
+        
+        uint basePrice = rentPrices[len - 1].mul(duration);
+        basePrice = basePrice.add(_premium(name, expires, duration));
+
+        uint ethPrice = uint(usdOracle.read());
+        return basePrice.mul(1e18).div(ethPrice);
     }
 
     /**
      * @dev Sets rent prices.
      * @param _rentPrices The price array. Each element corresponds to a specific
      *                    name length; names longer than the length of the array
-     *                    default to the price of the last element.
+     *                    default to the price of the last element. Values are
+     *                    in base price units, equal to one attodollar (1e-18
+     *                    dollar) each.
      */
     function setPrices(uint[] memory _rentPrices) public onlyOwner {
         rentPrices = _rentPrices;
         emit RentPriceChanged(_rentPrices);
     }
 
     /**
-     * @dev Returns the price to register or renew a name.
-     * @param name The name being registered or renewed.
-     * @param duration How long the name is being registered or extended for, in seconds.
-     * @return The price of this renewal or registration, in wei.
+     * @dev Sets the price oracle address
+     * @param _usdOracle The address of the price oracle to use.
      */
-    function price(string calldata name, uint /*expires*/, uint duration) view external returns(uint) {
-        uint len = name.strlen();
-        if(len > rentPrices.length) {
-            len = rentPrices.length;
-        }
-        require(len > 0);
-        uint priceUSD = rentPrices[len - 1].mul(duration);
+    function setOracle(DSValue _usdOracle) public onlyOwner {
+        usdOracle = _usdOracle;
+        emit OracleChanged(address(_usdOracle));
+    }
 
-        // Price of one ether in attodollars
+    /**
+     * @dev Returns the pricing premium in wei.
+     */
+    function premium(string calldata name, uint expires, uint duration) external view returns(uint) {
         uint ethPrice = uint(usdOracle.read());
+        return _premium(name, expires, duration).mul(1e18).div(ethPrice);
+    }
 
-        // priceUSD and ethPrice are both fixed-point values with 18dp, so we
-        // multiply the numerator by 1e18 before dividing.
-        return priceUSD.mul(1e18).div(ethPrice);
+    /**
+     * @dev Returns the pricing premium in internal base units.
+     */
+    function _premium(string memory name, uint expires, uint duration) internal view returns(uint) {
+        return 0;
```

Comment:
If you add this function just so that it compiles, may want to revert so that no one uses it by accident.

`
revert('Should not be called directly');
`

Reply:
No, it is called by `StablePriceOracle` if you use it instead of a derived class, because it never adds a premium.